src/test/scala/chiselTests/properties/PropertySpec.scala

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.properties
+
+import chisel3._
+import chisel3.properties.Property
+import chiselTests.ChiselFlatSpec
+
+class PropertySpec extends ChiselFlatSpec {
+  behavior.of("Property")
+
+  it should "fail to compile with unsupported Property types" in {
+    assertTypeError("""
+      class MyThing
+      val badProp = Property[MyThing]()
```

Comment:
Can you show the failing case where we try to `propOut := propIn` and you get the error message that its not legal? Such that in a subsequent PR it will be shown to now be legal?

Reply:
But that same test would show that you can indeed have IO ports of property types?

Reply:
Well on this PR `:=` is not even defined, so it will fail for that reason. Two PRs down the road there will be an example with both success and failure for `:=`.

Reply:
ok. i guess i thought this one was slated to land after #3442

Reply:
but in this PR i should be able to have IO of Input, Output properties? Can that be shown?

Reply:
> i guess i thought this one was slated to land after

These are currently independent, there will be a future change to actually make properties connectable.

> i should be able to have IO of Input, Output properties? Can that be shown?

Not in the current PR, since no actual types are being defined to work with Property (there are no typeclass instances for PropertyType). I'm trying to just stage these in small pieces with this piece being the initial boilerplate. In the next PR, you will be able to have an IO of Input or Output properties. If it would help, I can make draft PRs for the next two changes.

Reply:
ok, its fine. sounds like you are being cognizant of having the tests come in with the right change

Reply:
yep trying to break this up into small pieces with tests for what is supported in each piece. This PR is mostly boilerplate, but we can at least test one error.